unit_tests/src/test_connection_pool.cpp

```diff
@@ -379,6 +379,51 @@ TEST_CASE("connection pool")
             REQUIRE_NOTHROW(adm::client::modify_resource(conn, "demoResc", adm::resource_comments_property{""}));
         }
 
+        SECTION("refresh connection after adding/removing resource")
```

Comment:
Would it be worth making a test in which a resource is added and a different, older resource is removed? This would make the resource count appear to be the same on the connection fetch but the latest modify_time would be different due to the resource being added. Just a suggestion

Reply:
Sorry, to clarify, the timeline would look like this, as a minimal example:
1. Create resource A
2. Instantiate connection pool
3. Create resource B, remove resource A
4. Fetch a connection from the pool and assert that it has refreshed

Reply:
this seems a good thing to do.   and i also think the current code/plan will handle it just fine.  (also good).

Reply:
I'm confused lol. What are we testing?

My guess is you're talking about proving the connections are refreshed even when the latest mtime hasn't changed, but the resource count has due to the removal of an old resource. Is that correct?

Reply:
no... the count will stay the same, but the latest time is updated/newer.

this is just a complicated version of just updating a single resource.

Reply:
I see. We're talking about another client modifying the resources on the side.

Yes, definitely need tests for those cases.

Reply:
Yeah, like @trel said, it's not different from the modify case in essence but could catch some future issue should the logic change.

Reply:
I've added a new test to capture the behavior you mentioned.

Please review.